client/app/telecom/pack/xdsl/access/comfortExchange/pack-xdsl-access-comfort-exchange.controller.js

```diff
@@ -0,0 +1,81 @@
+import { COMFORT_EXCHANGE_TYPE_ERROR } from './pack-xdsl-access-comfort-exchange.constant';
+
+angular.module('managerApp').controller('XdslAccessComfortExchangeCtrl', class XdslAccessComfortExchangeCtrl {
+  /* @ngInject */
+  constructor($stateParams, $translate, OvhApiXdsl, TucToast, TucToastError) {
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.OvhApiXdsl = OvhApiXdsl;
+    this.TucToast = TucToast;
+    this.TucToastError = TucToastError;
+  }
+
+  /*= =====================================
+  =            INITIALIZATION            =
+  ====================================== */
+  $onInit() {
+    this.exchange = {
+      order: {
+        url: null,
+        id: null,
+        priceWithoutTax: null,
+        priceWithTax: null,
+        tax: null,
+      },
+      isSuccess: false,
+    };
+    this.isAvailable = false;
+    this.getListOpenedRMA();
+  }
+
+  getListOpenedRMA() {
+    this.isRMAOpened = false;
+    return this.OvhApiXdsl.RMA().v6().query({
+      xdslId: this.xdslId,
+    }).$promise.then((result) => {
+      if (result.length > 0) {
+        this.rmas = [];
+        this.isRMAOpened = true;
+        result.forEach(id => this.OvhApiXdsl.RMA().v6().getRMA({
+          xdslId: this.xdslId,
+        }, { id }).$promise.then((rma) => {
+          const addRma = {
+            creationDateTime: rma.creationDatetime,
+            equipmentReference: rma.equipmentReference,
+            id: rma.id,
+            newMerchandise: rma.newMerchandise,
+            status: rma.status,
+          };
+          this.rmas.push(addRma);
+        }));
+      }
+    });
+  }
+
+  comfortExchange() {
+    return this.OvhApiXdsl.Modem().v6().comfortExchange({
+      xdslId: this.xdslId,
+    }, {}).$promise.then((result) => {
+      this.exchange.isSuccess = true;
+      this.exchange.order.url = result.url;
```

Comment:
Do you have to respect this template ? 

If you don't have to, you can `_.assign(this.exchange, result)`